src/phoenix/experimental/evals/models/openai.py

```diff
@@ -202,14 +204,20 @@ def max_context_size(self) -> int:
         return context_size
 
     @property
-    def invocation_params(self) -> Dict[str, Any]:
+    def public_invocation_params(self) -> Dict[str, Any]:
         return {
             **({"engine": self.engine} if self._is_azure else {"model": self.model_name}),
             **self._default_params,
-            **self._credentials,
             **self.model_kwargs,
         }
 
+    @property
+    def invocation_params(self) -> Dict[str, Any]:
+        return {
+            **self.public_invocation_params,
+            **self._credentials,
+        }
```

Comment:
Suggestion: Do we need to store the credentials in a property? Let's avoid adding more properties if we can. The slimmer we keep the user-facing API, the less we have to document and the more flexibility we have to make changes.

I am not sure we need an `invocation_params` property at all, we might nix.

Reply:
clarified offline